### PR TITLE
Add Wagtail 2.3, Wagtail 2.7, and Django 2.2 in tox.ini

### PR DESCRIPTION
**Pull request by cwdavies**
Add support in tox.ini for unit testing with Wagtail 2.3 and Wagtail 2.7
Remove support in tox.ini for unit testing with Wagtail 2.0
Update Django from 2.0 to 2.2 in tox.ini

## Additions

- wag23
- wag27
- dj22

## Removed

- wag20
- dj20

## Testing

1. tox -e unittest-py36-dj111-wag23-slow
2. tox -e unittest-py36-dj111-wag27-slow
3. tox -e unittest-py36-dj22-wag23-slow
4. tox -e unittest-py36-dj22-wag27-slow

## Notes

Unit Tests are expected to fail, but will not cause Travis builds to fail
Tests will fail as code does not currently support Wagtail 2.x

## Checklist

- [X] PR has an informative and human-readable title
- [X] Changes are limited to a single goal (no scope creep)
- [X] Code can be automatically merged (no conflicts)
- [X] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Project documentation has been updated
- [X] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Browsers

- [ ] Chrome on desktop
- [ ] Firefox
- [ ] Safari on macOS
- [ ] Edge
- [ ] Internet Explorer 9, 10, and 11
- [ ] Safari on iOS
- [ ] Chrome on Android

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -21,12 +21,16 @@ envlist=lint-py{36}, unittest-py{36}-dj{111}-wag{113}-slow
 #   dj20:               Use Django 2.0
 #   wag113:             Use Wagtail 1.13
 #   wag20:              Use Wagtail 2.0
+#   wag23:              Use Wagtail 2.3
+#   wag27:              Use Wagtail 2.7
 #
 # These factors are expected to combine into the follow generative environments:
 #
 #   lint-py{36}
 #   unittest-py{36}-dj{111}-wag{113}-{fast,slow}
```
Comment (willbarton) [inline tox.ini]: It wasn't documented for `wag20`, but this should be a possible combination of factors:

```suggestion
 #   unittest-py{36}-dj{111}-wag{113,23}-{fast,slow}
```

File: tox.ini [hunk 2]
```diff
@@ -21,12 +21,16 @@ envlist=lint-py{36}, unittest-py{36}-dj{111}-wag{113}-slow
...
 #   unittest-py{36}-dj{20}-wag{20}-{fast,slow}
+#   unittest-py{36}-dj{20}-wag{23}-{fast,slow}
```
Comment (willbarton) [inline tox.ini]: These two lines can be combined:

```suggestion
#   unittest-py{36}-dj{20}-wag{23,27}-{fast,slow}
```

File: tox.ini [hunk 3]
```diff
@@ -21,12 +21,16 @@ envlist=lint-py{36}, unittest-py{36}-dj{111}-wag{113}-slow
...
+#   unittest-py{36}-dj{20}-wag{27}-{fast,slow}
```
Comment (willbarton) [inline tox.ini]:
```suggestion
```

File: tox.ini [hunk 4]
```diff
@@ -21,12 +21,16 @@ envlist=lint-py{36}, unittest-py{36}-dj{111}-wag{113}-slow
 #   dj20:               Use Django 2.0
 #   wag113:             Use Wagtail 1.13
 #   wag20:              Use Wagtail 2.0
```
Comment (willbarton) [inline tox.ini]: We can remove `wag20`, I think.
```suggestion
```

File: tox.ini [hunk 5]
```diff
@@ -36,6 +40,10 @@ envlist=lint-py{36}, unittest-py{36}-dj{111}-wag{113}-slow
 #   tox -e unittest-py36-dj111-wag113-slow
 #   tox -e unittest-py36-dj20-wag20-fast
```
Comment (willbarton) [inline tox.ini]:
```suggestion
 #   tox -e unittest-py36-dj111-wag23-fast
```

File: tox.ini [hunk 6]
```diff
@@ -53,6 +61,8 @@ deps=
     dj20:               Django>=2.0,<2.1
     wag113:             wagtail>=1.13,<1.14
     wag20:              wagtail>=2.0,<2.1
```
Comment (willbarton) [inline tox.ini]:
```suggestion
```

File: tox.ini [hunk 7]
```diff
@@ -36,6 +40,10 @@ envlist=lint-py{36}, unittest-py{36}-dj{111}-wag{113}-slow
 #   tox -e unittest-py36-dj111-wag113-slow
 #   tox -e unittest-py36-dj20-wag20-fast
 #   tox -e unittest-py36-dj20-wag20-slow
```
Comment (willbarton) [inline tox.ini]:
```suggestion
```

**Comment by cwdavies** [comment]
Primarily made changes to remove Wagtail 2.0. Updated tox.ini to combine Wagtail 2.3 with Django 1.11. Wagtail 2.7 is only combined with Django 2.0.

**Comment by willbarton** [comment]
This looks good. Because we should be close to being able to use GitHub Actions again, I'm going to update #5302. Once this is merged, I'll try to make it so it's able to run these as a matrix and allow the wag23, wag27, and dj22 factors to fail for now.

**Pull request merged**